Composer/packages/client/src/components/ManageService/ManageService.tsx

```diff
@@ -95,7 +96,7 @@ export const ManageService = (props: ManageServiceProps) => {
   const [availableSubscriptions, setAvailableSubscriptions] = useState<Subscription[]>([]);
   const [subscriptionsErrorMessage, setSubscriptionsErrorMessage] = useState<string>();
   const [keys, setKeys] = useState<KeyRec[]>([]);
-
+  const [userProvidedTokens, setUserProvidedTokens] = useState<boolean>(false);
```

Comment:
what's the difference between userProvidedTokens and !!token? (i.e., token has value)

Reply:
we need to know if it's in dev mode and user provided the tokens, instead of calling userShouldProvideTokens(), I am using this state.